mongomock/aggregate.py

```diff
@@ -256,6 +263,40 @@ def _handle_comparison_operator(self, operator, values):
             'aggregation pipeline, it is currently not implemented '
             ' in Mongomock.' % operator)
 
+    def _handle_misc_operator(self, operator, values):
+        if operator == '$toString':
+            parsed = self.parse(values)
+            return str(parsed) if parsed is not None else None
+            # This should never happen: it is only a safe fallback if something went wrong.
+        raise NotImplementedError(  # pragma: no cover
+            "Although '%s' is a valid operator for the aggregation "
+            'pipeline, it is currently not implemented  in Mongomock.' % operator)
+
+    def _handle_string_operator(self, operator, values):
+        if operator == '$toLower':
+            return str(self.parse(values)).lower()
+        if operator == '$toUpper':
+            return str(self.parse(values)).upper()
+        if operator == '$concat':
+            return ''.join([str(self.parse(value)) for value in values])
+        if operator == '$substr':
+            assert len(values) == 3, 'substr must have 3 items'
+            string = str(self.parse(values[0]))
+            first = self.parse(values[1])
+            length = self.parse(values[2])
+            if string is None or first < 0:
+                return ''
+            second = len(string) if length < 0 else first + length
```

Comment:
Please add a test case for negative length.

Reply:
forgot it on push - so added the test and pushed again.